command_line/main.py

```diff
@@ -0,0 +1,283 @@
+import argparse
+
+from methods import Method
+from models import Phenotype, Experiment
+
+from .parser import Parser, Argument
+from .types import Slice, one_of, Indices, dsv, Range
+from .method_parser import MethodParser
+
+
+class PhenotypeFactory(Parser):
+    """Provide {parser_name} samples. Requires a file (or files) with samples.
+
+     The files should come in Delimiter Separated Values format
+     (like .csv or .tsv). The default delimiter is a tab character.
+     First column of each file should contain gene identifiers.
+
+     To use only a subset of samples from files(s) specify column numbers
+     (--columns) or sample names (--samples) of desired samples.
+     """
+
+    files = Argument(
+        type=argparse.FileType('r'),
+        # at least one file is always required
+        nargs='+',
+        optional=False
+    )
+
+    name = Argument(help='Your custom name for this set of samples.')
+
+    samples = Argument(
+        type=dsv(str),
+        nargs='*',
+        as_many_as=files,
+        help='Names of samples (columns) to be extracted from the file. '
+             'Sample names are determined from the first non-empty row. '
+             'Use comma to separate samples. '
+             'Samples for each of files should be separated by space.'
+    )
+
+    columns = Argument(
+        # we want to handle either ":4", "5:" or even "1,2,3"
+        type=one_of(Slice, Indices, Range),
+        # user may (but do not have to) specify columns
+        # to be extracted from given file(s).
+        nargs='*',
+        as_many_as=files,
+        help='Columns to be extracted from files: '
+             'either a comma delimited list of 0-based numbers (e.g. 0,2,3) '
+             'or a range defined using Python slice notation (e.g. 3:10). '
+             'Columns for each of files should be separated by space.'
+    )
+
+    delimiter = Argument(
+        default='\t',
+        help='Delimiter of the provided file(s). Default: tabulation mark.'
+    )
+
+    # TODO: do we usually have sample names in headers (the default assumes that we have)
+    header = Argument(
+        nargs='*',
+        type=one_of(int, str),
+        as_many_as=files,
+        default=lambda file_object: 0,
+        help='Defines how the sample names should be created. '
+             'Provide a number to specify which line should be used '
+             'to extract names for samples. Please remember that '
+             'empty lines will be skipped. If your file has no row '
+             'with sample names, provide a string to be used as a '
+             'prefix for naming consecutive samples. '
+             'For example, `--header cancer` will lead to naming '
```

Comment:
I like the simplicity of this sample naming when there is no row with sample names.
As for the default I think that the way it is now would be the best. Moreover, according to this:
https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#TXT:_Text_file_format_for_expression_dataset_.28.2A.txt.29
we should expect the row with sample names AND ! even the additional description column.
I would like to discuss the optional description column: should we add this possibility (more user-friendly!) and in which way (automatic detection or user-provied argument?) or do we just leave it like it is now? What do you think?

Reply:
Certainly we need to to something with that.

I think that an additional
```python
description_column = Argument(short="d", type=positive_int, default=None, help="Index of column with descriptions of genes.")
```
would be a good solution. I am not sure about automatic detection, but we certainly should at least warn the user that there is a column "description".

The good thing is that we can pass a multi-index to `index_col` so it should be easy to implement.

Reply:
Is this possible that the user provides two files with descriptions (one for control second for case, or both for control, whatever) and the descriptions differ among the files? Is this an error of the user or should we support such a case?

Reply:
I added the `--description_column`, which is kind of solution for now. It needs to be further refined. For example we have to test the behavior for `-d 0` (description and gene symbol columns are the same, what's now?). Or maybe we should only have a switch `--description_column`, without an option to specify which column it is (assuming it is column number 1).

Reply:
Nice, that was fast! I would rather go with a switch option, but let's wait for other opinions. In the case of differing descriptions... For now, it seems that this information will be useless for us. So it depends on whether we want to give a user some way to save output data with descriptions provided. If yes then we have to decide what to do and I think that it would be ok to "fuse" he different descriptions i.e. 'description1/description2'